presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/IOPlanPrinter.java

```diff
@@ -0,0 +1,402 @@
+                throw new IllegalStateException(format("Unknown WriterTarget subclass %s", writerTarget.getClass().getSimpleName()));
+            }
+            return processChildren(node, context);
+        }
+
+        private List<ColumnConstraint> parseConstraints(TableHandle tableHandle, TupleDomain<ColumnHandle> constraint)
+        {
+            checkArgument(!constraint.isNone());
+            ImmutableList.Builder<ColumnConstraint> columnConstraints = ImmutableList.builder();
+            for (Map.Entry<ColumnHandle, Domain> entry : constraint.getDomains().get().entrySet()) {
+                ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, entry.getKey());
+                columnConstraints.add(new ColumnConstraint(columnMetadata.getName(), columnMetadata.getType(), parseDomain(entry.getValue())));
+            }
+            return columnConstraints.build();
+        }
+
+        private FormattedDomain parseDomain(Domain domain)
+        {
+            ImmutableList.Builder<FormattedRange> formattedRanges = ImmutableList.builder();
+            Type type = domain.getType();
+
+            domain.getValues().getValuesProcessor().consume(
+                    ranges -> formattedRanges.addAll(
+                            ranges.getOrderedRanges().stream()
+                                    .map(range -> new FormattedRange(formatMarker(range.getLow()), formatMarker(range.getHigh())))
+                                    .collect(toImmutableList())),
+                    discreteValues -> formattedRanges.addAll(
+                            discreteValues.getValues().stream()
+                                    .map(value -> getVarcharValue(type, value))
+                                    .map(value -> new FormattedMarker(Optional.of(value), EXACTLY))
+                                    .map(marker -> new FormattedRange(marker, marker))
+                                    .collect(toImmutableList())),
+                    allOrNone -> {
+                        throw new IllegalStateException("Unreachable AllOrNone consumer");
+                    });
+
+            return new FormattedDomain(domain.isNullAllowed(), formattedRanges.build());
+        }
+
+        private FormattedMarker formatMarker(Marker marker)
+        {
+            if (!marker.getValueBlock().isPresent()) {
+                return new FormattedMarker(Optional.empty(), marker.getBound());
+            }
+            return new FormattedMarker(Optional.of(getVarcharValue(marker.getType(), marker.getValue())), marker.getBound());
+        }
+
+        private String getVarcharValue(Type type, Object value)
+        {
+            if (type instanceof VarcharType) {
+                return ((Slice) value).toStringUtf8();
+            }
+            if (type instanceof TinyintType || type instanceof SmallintType || type instanceof BigintType || type instanceof DoubleType) {
+                return String.valueOf(value);
```

Comment:
I would prefer `((Long) value).toString`. `String.valueOf` is unnecessarily generic here.